File: python-api/src/api/handlers/polls/getPollByTitle/getPollByTitleHandler.py
try:
    import unzip_requirements
except ImportError:
    pass

import boto3
import os
import simplejson as json

dynamodb_client = boto3.resource('dynamodb')
TABLE_NAME = os.environ['POLLS_TABLE_NAME']
DYNAMO_POLLS_ID = os.environ['POLLS_ID']

def create_response(code, message):
  return {
    'statusCode': code,
    'headers': {'Content-Type': 'application/json'},
    'body': json.dumps(message)
  }

def getPollByTitle(event, context):
    title = event['queryStringParameters']['title']

    params = {
        'IndexName': 'SearchPollByTitle',
        'KeyConditionExpression': 'title = :v AND begins_with(PK, :v1)',
        'ExpressionAttributeValues': {
            ':v': title,
            ':v1': 'POLL'
        }
    }

    try:
        table = dynamodb_client.Table(TABLE_NAME)
        response = table.query(**params)
    except Exception as ex:
        return create_response(500, str(ex))

    print('response: ', response)

    if response['ResponseMetadata']['HTTPStatusCode'] != 200:
      return create_response(500, 'Error getting the poll')

    if not response['Items'] or len(response['Items']) != 1:
        return create_response(500, { 'message': 'Poll not found' })

    poll = response['Items'][0]

    return create_response(200, poll)

File: nodejs-api/src/api/handlers/polls/getPollByTitle/getPollByTitleHandler.js
'use strict';

const AWS = require('aws-sdk')
const db = new AWS.DynamoDB.DocumentClient({})

const tableName = process.env.POLLS_TABLE_NAME

const response = (statusCode, message) => {
    return {
        statusCode: statusCode,
        body: JSON.stringify(message)
    }
}

const getPollByTitle = async (event) => {
    const title = event.queryStringParameters.title
    console.log('title: ', title)
  
    const params = {
      TableName: tableName,
      IndexName: 'SearchPollByTitle',
      KeyConditionExpression: 'title = :v AND begins_with(PK, :v1)',
      ExpressionAttributeValues: {
        ':v': title,
        ':v1': 'POLL'
      }
    }
  
    try {
      const pollSearch = await db.query(params).promise()
      console.log('pollSearch', pollSearch)
  
      if (!pollSearch.Items || pollSearch.Items.length !== 1) {
        return response(500, {
          message: "Poll not found"
        })
      }
  
      const poll = pollSearch.Items[0]
      
      return response(200, poll)
    } catch (err) {
      console.log('err: ', err)
      return response(err.statusCode, err)
    }
}

module.exports = {
    getPollByTitle
}